cnd/src/network/comit_ln.rs

```diff
@@ -306,18 +306,55 @@ impl NetworkBehaviourEventProcess<announce::behaviour::BehaviourOutEvent> for Co
     fn inject_event(&mut self, event: announce::behaviour::BehaviourOutEvent) {
         match event {
             announce::behaviour::BehaviourOutEvent::ReceivedAnnouncement { peer, mut io } => {
+                // Check if there are any errors before modifying the hash-map.
+                match self.swaps_waiting_for_announcement.get(&io.swap_digest) {
```

Comment:
Once https://github.com/comit-network/comit-rs/pull/2479 is merged, it should be easy to have an integration test for this :)